stdlib/http/src/main/java/org/ballerinalang/net/http/WebSocketDispatcher.java

```diff
@@ -108,12 +121,76 @@ static void dispatchTextMessage(WebSocketOpenConnectionInfo connectionInfo,
         List<ParamDetail> paramDetails = onTextMessageResource.getParamDetails();
         BValue[] bValues = new BValue[paramDetails.size()];
         bValues[0] = connectionInfo.getWebSocketEndpoint();
-        bValues[1] = new BString(textMessage.getText());
-        if (paramDetails.size() == 3) {
-            bValues[2] = new BBoolean(textMessage.isFinalFragment());
+        boolean finalFragment = textMessage.isFinalFragment();
+        BType dataType = paramDetails.get(1).getVarType();
+        int dataTypeTag = dataType.getTag();
+        if (dataTypeTag == TypeTags.STRING_TAG) {
+            bValues[1] = new BString(textMessage.getText());
+            if (paramDetails.size() == 3) {
+                bValues[2] = new BBoolean(textMessage.isFinalFragment());
+            }
+            Executor.submit(onTextMessageResource, new WebSocketResourceCallableUnitCallback(webSocketConnection),
+                            null, null, bValues);
+        } else if (dataTypeTag == TypeTags.JSON_TAG || dataTypeTag == TypeTags.RECORD_TYPE_TAG ||
+                dataTypeTag == TypeTags.XML_TAG || dataTypeTag == TypeTags.ARRAY_TAG) {
+            if (finalFragment) {
+                aggregateString += textMessage.getText();
+                dispatchReourceWithAggregatedData(webSocketConnection, onTextMessageResource, bValues, dataType);
+                aggregateString = "";
```

Comment:
What's the usage of this aggregateString? We are concat a value in line 137 but without using it, assign empty in above line?

Reply:
It is used in the `dispatchReourceWithAggregatedData` method. I have a new method because sonar complains about having a single long method. `aggregateString` is global because it aggregates string based on the `finalFragment` value.